### PR TITLE
Add endpoint that returns commit sha given a valid reference

### PR DESCRIPTION
**Pull request by carlosdiazh**
Adds a new endpoint that returns the specific commit sha of a given branch name or tag.

### CODE REVIEW

File: server/src/main.rs [hunk 1]
```diff
@@ -367,4 +395,24 @@ mod tests {
             "[\"file-a\",\"file-b\",\"file-c\",\"nested-dir\"]"
         );
     }
+
+    // resolve tests
+
+    #[test]
+    fn resolve_ref_with_empty_repo() {
```
Comment (nathankleyn) [inline server/src/main.rs]: Name of this tests isn't quite correct.

Comment (nathankleyn) [reply server/src/main.rs]: And will need a test to check empty repo handling I think too.

File: server/src/main.rs [hunk 2]
```diff
@@ -367,4 +395,24 @@ mod tests {
...
+        assert_test_server_responds_with!(
+            "/repos/fixtures/resolve?reference=467e981f94686d7a1db395f8acfd3cf7e7adfcd3",
+            200,
+            "467e981f94686d7a1db395f8acfd3cf7e7adfcd3"
+        )
+    }
+
+    #[test]
+    fn resolve_ref_with_invalid_ref() {
```
Comment (nathankleyn) [inline server/src/main.rs]: Missng some happy path tests here when you give a branch or tag I think, to make sure you get a valid SHA back.

**Comment by nathankleyn** [comment]
There’s some legit chippy failures on this build which need to be fixed first before merging — but code is otherwise 👍👍!
